File: Pizzeria_V04/app/src/main/java/com/example/pizzeria_v04/GUI_LOGIN.java
package com.example.pizzeria_v04;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Toast;

public class GUI_LOGIN extends AppCompatActivity
{

    EditText edtUsuLogin;
    EditText edtPassLogin;
    Button btnLogin;
    CheckBox checkLogin;

    String usuario, password;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_gui__login);

        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);

        // Asociamos los id's
        edtUsuLogin = findViewById(R.id.idEdtUsuLogin);
        edtPassLogin = findViewById(R.id.idEdtPassLogin);
        btnLogin = findViewById(R.id.idBtnLogin);
        checkLogin = findViewById(R.id.idCheckLogin);

        // Evento btn SIGUIENTE
        btnLogin.setOnClickListener(new View.OnClickListener()
        {
            @Override
            public void onClick(View v)
            {

                // Guardamos los datos introducidos por el usuario
                usuario = edtUsuLogin.getText().toString().toLowerCase();
                password = edtPassLogin.getText().toString().toLowerCase();

                // Leemos datos del SharedPreferences
                SharedPreferences sp = getSharedPreferences(usuario+"data", MODE_PRIVATE);
                String usu = sp.getString("usuario", "Error usuario");
                String pass = sp.getString("pass", "Error contraseña");

                // Comprobar que los datos estén en Registrados
                if(usuario.equals(usu) && password.equals(pass))
                {
                    Intent intent = new Intent(getApplicationContext(), GUI_PANEL_OPCIONES.class);
                    intent.putExtra("mandamosPassLogin", pass); //Enviamos contraseña al PANEL DE OPCIONES
                    intent.putExtra("mandamosNombrFichDelLogin", usuario+"data.xml");
                    startActivity(intent);
                }
                else
                {
                    getToast("Compruebe nombre y contraseña");
                }
            }
        });
    }

    // Método TOAST
    public void getToast(String msg)
    {
        Toast.makeText(getApplicationContext(), msg, Toast.LENGTH_LONG).show();
    }

}

File: Pizzeria/app/src/main/java/com/example/pizzeria/Gui_Login.kt
package com.example.pizzeria

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.Toast

class Gui_Login : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_gui__login)

        val edtUsuario = findViewById<EditText>(R.id.idEdtUsuarioLogin) as EditText
        val edtPass = findViewById<EditText>(R.id.idEdtPassLogin) as EditText
        val btnLogin = findViewById<Button>(R.id.idBtnLogin) as Button
        val checkingBox = findViewById<CheckBox>(R.id.idCheckLogin) as CheckBox

        // Evento del BOTÓN
        btnLogin.setOnClickListener {
            // Debemos comprobar que el usuario esté registrado ( SharedPreferences del Usuario en Registro )
            val name = edtUsuario.text.toString()
            val pass = edtPass.text.toString()
            val nameFile = name+"dataRegister"

            if(fileExist(nameFile))
            {
                getToast("Usuario existe...")
            }
        }

        // Evento del CheckBox
        checkingBox.setOnClickListener {

            // Debemos comprobar que el usuario esté registrado ( SharedPreferences del Usuario en Registro )
            val name = edtUsuario.text.toString()
            val pass = edtPass.text.toString()
            val nameFile = name+"dataRegister"

            if(fileExist(nameFile))
            {
                getToast("Usuario existe...")
            }

            /*if(checkingBox.isChecked) // Si usuario hace click en CheckBox
            {
                // Guardamos los datos en un xml SharedPreferences
                val sharedPref: SharedPreferences = getSharedPreferences(edtUsuario.text.toString() + "dataLogin", Context.MODE_PRIVATE)
                val editor = sharedPref.edit()

                editor.putString("name", edtUsuario.text.toString())
                editor.putString("password", edtPass.text.toString())

                var datosUser = sharedPref.getString("name", "defautVALUE")
                getToast(datosUser.toString())
            }
*/

        }
    }

    fun fileExist(file: String): Boolean
    {
        var exist: Boolean = false;

        val sharedPref: SharedPreferences = getSharedPreferences(file, Context.MODE_PRIVATE)

        var datosUser = sharedPref.getString("name", "defautVALUE")
        var datosEmail = sharedPref.getString("email", "defalutVALUE")
        var datosPass = sharedPref.getString("password", "defaultValue")

        if(!datosUser.toString().equals("defautVALUE") &&
                !datosPass.toString().equals("defautVALUE"))
        {
            exist = true;
        }
        return exist;
    }

    fun getToast(msg:String)
    {
        Toast.makeText(this, msg, Toast.LENGTH_LONG).show()
    }
}
